### PR TITLE
refactor: move `runBonjour`and `runOpen` utils  to `Server`

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [X] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Already present.

### Motivation / Use-Case

Inline utils with the server.

### Breaking Changes
None

### Additional Info
After this

### CODE REVIEW

**Review by anshumanv — DISMISSED** [review]

**Review by rishabh3112 — APPROVED** [review]

File: lib/Server.js [hunk 1]
```diff
@@ -538,6 +538,110 @@ class Server {
...
+      subtypes: ['webpack'],
+      ...options.bonjour,
+    });
+
+    process.on('exit', () => {
+      bonjour.unpublishAll(() => {
+        bonjour.destroy();
+      });
+    });
+  }
```
Comment (alexander-akait) [inline lib/Server.js:559]: It should not be static function, it is not public API

File: lib/Server.js [hunk 2]
```diff
@@ -538,6 +538,110 @@ class Server {
...
+                        )}" arguments`
+                      : ''
+                  }`
+                : ''
+            }. If you are running in a headless environment, please do not use the "open" option or related flags like "--open", "--open-target", and "--open-app".`
+          );
+        });
+      })
+    );
+  }
```
Comment (alexander-akait) [inline lib/Server.js:643]: Ideally will be great to inline inside code

Comment (alexander-akait) [reply lib/Server.js:643]: Also I think we need better name...

Comment (snitin315) [reply lib/Server.js:643]: Maybe `openBrowser`?

Comment (alexander-akait) [reply lib/Server.js:643]: Yep sounds good

File: lib/Server.js [hunk 3]
```diff
@@ -538,6 +538,110 @@ class Server {
...
+
+    if (Array.isArray(options)) {
+      options.forEach((item) => {
+        openTasks.push(...getOpenTask(item));
+      });
+    } else {
+      openTasks.push(...getOpenTask(options));
+    }
+
+    return Promise.all(
```
Comment (alexander-akait) [inline lib/Server.js:609]: I think we don't need `return` here, because we don't use returned result :smile:

Comment (snitin315) [reply lib/Server.js]: Yes, good catch.

**Review by rishabh3112 — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Comment by alexander-akait** [comment]
@snitin315 I will solve conflicts, anyway we should remove more utils, some of them we can move in normalize, some as static methods

**Comment by alexander-akait** [comment]
Found more cases, WIP

**Comment by alexander-akait** [comment]
@snitin315 Feel free to remove more utils, ideally we should keep only two utils after refactor:
- `normalizeOptions.js` (we will move this as method in future)
- `DevServerPlugin.js` (will be removed when we will have plugin support here)

Both will be remove in the one future PR

Also because we already do many breaking change and will write migration guide, make sense to remove `findPort.js` and say in docs what you need the latest version of `webpack-cli` to using `webpack-dev-server`

**Comment by snitin315** [comment]
Will take care of it 👍

**Comment by alexander-akait** [comment]
@snitin315 But let's do it in separate PR :+1: I want to merge this and continue

**Pull request merged**